http-verbs-common/src/main/scala/uk/gov/hmrc/http/client/HttpClientV2.scala

```diff
@@ -77,5 +77,5 @@ trait RequestBuilder {
   /** `withBody` should be called rather than `transform(_.withBody)`.
     * Failure to do so will lead to a runtime exception
     */
-  def withBody[B : BodyWritable : TypeTag](body: B): RequestBuilder
+  def withBody[B : BodyWritable : TypeTag](body: B)(implicit ec: ExecutionContext): RequestBuilder
```

Comment:
I think providing an execution context here is preferable to passing around the `Promise` internally, so that it is converted to a `Future` in the `execute` step.